Automattic/jetpack
# Odyssey: store the timestamp when enabling / disabling new Stats

## Proposed changes:
The PR proposes to 

- Add new stats options to store the timestamp when Odyssey is enabled / disabled, so that we could implement the "This prompt should be shown a month after the customer has opted out of Odyssey Stats" in #28729.
- Add `collapse_nudges` to whitelist, otherwise it'd be cleared on saving.
- Renamed stats option `enable_calypso_stats` to `enable_odyssey_stats`
- *Default `enable_odyssey_stats` to true, meaning for Jetpack sites old or new, Odyssey would be the default experience‼️* 

### Other information:


## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Go to `/wp-admin/admin.php?page=jetpack#/traffic` and find section `Jetpack Stats`
* Toggle `Enable the new Jetpack Stats experience New`
* Open a MySQL querying tool as you like, for example PHPMyAdmin
* Run SQL: `SELECT * FROM `wp_options` WHERE `option_name` LIKE 'stats_options'`
* Unserialize `option_value` with [online tool](https://www.unserialize.com/) or locally with PHP CLI
* Ensure 'collapse_nudges', 'enable_calypso_stats', 'calypso_stats_last_enabled_at', 'calypso_stats_last_disabled_at' changes as expected.
